### PR TITLE
Dark Skies Above: Surveillance drone scan and reinforcements

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
Features "Dark Skies Above: Surveillance drone scan and reinforcements"

#### Purpose of change
Make alien surveillance drones *scary*, by having them scan nearby characters for weapons and summon the new alien skirmisher monster.

#### Describe the solution
Co-authored by @ephemeralstoryteller 

This builds on #48155 and includes it.

Alien drones get a new special attack, which is unfortunately hard-coded.  This special attack checks nearby, visible characters for weapons, no more than once every 10 seconds.  The timer is individual per character.  If the character is wearing or holding a gun, a counter (again, individual to the character) gets incremented.  If the counter hits 3, a hard-coded timed event is set.

The timed event goes off 5-10 seconds later, and either uses a spell to place 3-4 alien skirmishers on the local map, or uses a map extra to place them near where the drone was if the avatar has moved the drone away from the reality bubble.

Alien skirmishers are nasty, melee focused opponents with LEAP and a taser-like electrical attack.

#### Testing
I confirmed that the alien drones would scan for weapons and place the skirmishers, whether or not the avatar was present.

### CODE REVIEW

File: data/mods/Dark-Skies-Above/monsters/alien_robots.json [hunk 1]
```diff
@@ -50,6 +75,7 @@
     "vision_night": 5,
     "special_attacks": [
       [ "TAZER", 12 ],
+      [ "DSA_DRONE_SCAN", 5 ],
```
Comment (ephemeralstoryteller) [inline data/mods/Dark-Skies-Above/monsters/alien_robots.json:78]: My only concern with giving the seeker the scan is that it's already a somewhat dangerous attacker when angered. That said, it certainly does make them a much more dangerous enemy, so it's just a consideration.

Comment (ephemeralstoryteller) [inline data/mods/Dark-Skies-Above/monsters/alien_robots.json:78]: My only concern with giving the seeker the scan is that it's already a somewhat dangerous attacker when angered. That said, it certainly does make them a much more dangerous enemy, so it's just a consideration.

Comment (mlangsdorf) [reply data/mods/Dark-Skies-Above/monsters/alien_robots.json:78]: Having a drone show up at all should cause an unprepared survivor to run - and they have at least 35 seconds to escape the area without immediate consequences.

I do need to add it to the other drone so they're both worrisome.

Comment (mlangsdorf) [reply data/mods/Dark-Skies-Above/monsters/alien_robots.json:78]: And I did, it was just a single line special attack so I didn't see it.

File: src/monattack.cpp [hunk 2]
```diff
@@ -5812,3 +5812,92 @@ bool mattack::speaker( monster *z )
...
 }
+
+bool mattack::dsa_drone_scan( monster *z )
+{
+    z->moves -= 100;
+    constexpr int scan_range = 30;
+    // Select a target: the avatar or a nearby NPC.  Must be visible and within scan range
+    Character *target = &get_player_character();
+    Character &you = get_player_character();
+    bool avatar_in_range = z->posz() == target->posz() && z->sees( target->pos() ) &&
```
Comment (Night-Pryanik) [inline src/monattack.cpp:5823]: I understand this is a rare case, but that means that the drone won't be able to detect player from the roof or second floor?

Comment (mlangsdorf) [reply src/monattack.cpp:5823]: 3DFoV is still so messed up that I'm tempted to keep this for now.  I guess the `sees()` should catch any wackiness like looking through the ceiling or floor.

**Comment by actual-nh** [comment]
This unfortunately needs String Freeze (as in targeting at 0.F-dev).

**Comment by jbytheway** [comment]
I think this relies on a harvest entry `dks_alien_cyborg` which isn't defined (judging by recent CI failures).

**Comment by ephemeralstoryteller** [comment]
Good catch, I forgot to replace `dks_alien_cyborg` with `dks_alien_hcyborg`.

**Comment by jbytheway** [comment]
OK, I PRed that fix in #48478.

**Comment by ephemeralstoryteller** [comment]
Awesome, thank you!

**Pull request merged**